direct/tpm2/policy_test.go

```diff
@@ -281,6 +282,189 @@ func TestPolicyOrUpdate(t *testing.T) {
 	}
 }
 
+func TestPolicyPCRUpdate(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	// Use a trial session to calculate this policy
+	sess, cleanup2, err := PolicySession(thetpm, tpm.AlgSHA1, 16, Trial())
+	if err != nil {
+		t.Fatalf("setting up policy session: %v", err)
+	}
+	defer func() {
+		t.Helper()
+		if err := cleanup2(); err != nil {
+			t.Errorf("cleaning up policy session: %v", err)
+		}
+	}()
+
+	PCRs, err := CreatePCRSelection([]int{0, 1, 2, 3, 7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	selection := tpml.PCRSelection{
+		PCRSelections: []tpms.PCRSelection{
+			{
+				Hash:      tpm.AlgSHA1,
+				PCRSelect: PCRs,
+			},
+		},
+	}
+
+	pcrRead := PCRRead{
+		PCRSelectionIn: selection,
+	}
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to read PCRSelection")
+	}
+
+	var expectedVal []byte
+	for _, digest := range pcrReadRsp.PCRValues.Digests {
+		expectedVal = append(expectedVal, digest.Buffer...)
+	}
+	t.Logf("expectedVal=%x", expectedVal)
+
+	// Hash algorithm must match the one in PolicySession.
+	expectedDigest := sha1.Sum(expectedVal)
+	t.Logf("expectedDigest=%x", expectedDigest)
+
+	wrongCase := append(expectedVal, []byte("wrongCase")...)
+	wrongDigest := sha1.Sum(wrongCase)
+
+	tests := []struct {
+		desc           string
+		expectedDigest []byte
+		expectedResult bool
+	}{
+		{desc: "success: expectedDigest passed in expectedDigest", expectedDigest: expectedDigest[:], expectedResult: true},
+		{desc: "fail: wrongDigest passed in expectedDigest", expectedDigest: wrongDigest[:], expectedResult: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			policyPCR := PolicyPCR{
+				PolicySession: sess.Handle(),
```

Comment:
We probably should call PolicyRestart (or just flush and re-create the session) across iterations of this loop. If the second case was a positive-result case, it'd fail because `PolicyPCR(x) ⋅ PolicyPCR(y)` [where `⋅` represents the implicit AND of chaining multiple policy commands in one session] is a different policy than `PolicyPCR(y)` by itself

Reply:
Resolved, but ran into trouble when passing in the wrong policy. Ended up deciding to have the policy get executed first then change the digest prior to the policy calculator.